### PR TITLE
Implement GetMentorsByInternshipId endpoint

### PR DESCRIPTION
**Pull request by Xodoc**

### CODE REVIEW

File: InternshipBE/BL/Interfaces/IUserService.cs [hunk 1]
```diff
@@ -8,5 +9,7 @@ public interface IUserService
         Task<string> AuthenticateAsync(string email, string password);
 
         Task<UserDTO> GetUserInfoByUserNameAsync(string userName);
+
+        Task<List<UserDTO>> GetMentorsByInternshipId(int id);
```
Comment (romanstat) [inline InternshipBE/BL/Interfaces/IUserService.cs]: Async

File: InternshipBE/WebApi/Controllers/UserController.cs [hunk 2]
```diff
@@ -24,5 +24,11 @@ public async Task<IActionResult> GetUserInfo()
 
             return Ok(await _userService.GetUserInfoByUserNameAsync(userName));
         }
+
+        [HttpGet("getAllUsersByInternship")]
```
Comment (romanstat) [inline InternshipBE/WebApi/Controllers/UserController.cs]: [HttpGet("getAllUsersByInternship")], but GetMentorsByInternshipId???

File: InternshipBE/BL/Services/UserService.cs [hunk 3]
```diff
@@ -101,5 +112,21 @@ private async Task<User> GetUserByUserNameAsync(string userName)
...
+            {
+                var role = await GetUserRoleAsync(user);
+
+                if (role == RoleType.Mentor.ToString())
+                {
+                    mentors.Add(user);
+                }
+            }
+
+            return mentors;
```
Comment (romanstat) [inline InternshipBE/BL/Services/UserService.cs]: reimplement with repository

File: InternshipBE/DAL/Repositories/UserRepository.cs [hunk 4]
```diff
@@ -1,13 +1,64 @@
...
+            {
+                var role = await GetUserRoleAsync(user);
+
+                if (role == RoleType.Mentor.ToString())
+                {
+                    mentors.Add(user);
+                }
+            }
+
+            return mentors;
```
Comment (romanstat) [inline InternshipBE/DAL/Repositories/UserRepository.cs]: need refactor

File: InternshipBE/DAL/Repositories/UserRepository.cs [hunk 5]
```diff
@@ -1,13 +1,64 @@
...
+        }
+
+        public async Task<string> GetUserRoleAsync(User user)
```
Comment (romanstat) [inline InternshipBE/DAL/Repositories/UserRepository.cs]: public methods first than private

File: InternshipBE/BL/Services/UserService.cs [hunk 6]
```diff
@@ -77,16 +79,20 @@ public async Task<UserDTO> GetUserInfoByUserNameAsync(string userName)
...
+            string role = "Mentor";
+
+            foreach (var mentor in usersDTO)
             {
-                throw new ArgumentNullException(nameof(roles), "roles is null");
+                mentor.RoleType = Enum.Parse<RoleType>(role);
             }
 
-            return roles.Single();
+            return usersDTO;
```
Comment (romanstat) [inline InternshipBE/BL/Services/UserService.cs]: need refactor

File: InternshipBE/DAL/Extensions/ModelBuilderExtensions.cs [hunk 7]
```diff
@@ -96,7 +96,8 @@ public static ModelBuilder FillUsers(this ModelBuilder builder)
 				new User
 				{
 					Id = Guid.NewGuid().ToString(),
-					Position = "BA",
+                    RoleType = RoleType.Hr,
+                    Position = "BA",
 					BestContactTime = DateTime.UtcNow,
 					UserName = userName1,
 					NormalizedUserName = userName1.ToUpper(),
```
Comment (romanstat) [inline InternshipBE/DAL/Extensions/ModelBuilderExtensions.cs:103]: align

File: InternshipBE/DAL/Repositories/UserRepository.cs [hunk 8]
```diff
@@ -1,13 +1,31 @@
...
-		{
-		}
-	}
+    {
+
+        public UserRepository(ApplicationDbContext context) : base(context)
+        {
+        }
+
+        public async Task<List<User>> GetMentorsByInternshipIdAsync(int id)
```
Comment (romanstat) [inline InternshipBE/DAL/Repositories/UserRepository.cs]: GetSpecificUsersByInternshipIdAsync(id, RoleType? roleType = null)

File: InternshipBE/DAL/Repositories/UserRepository.cs [hunk 9]
```diff
@@ -1,13 +1,31 @@
...
+        {
+            var internship = await _context.Internships.AsNoTracking().Include(x => x.Users).FirstOrDefaultAsync(x => x.Id == id);
+
+            var roles = _context.Users.Where(x => x.RoleType == RoleType.Mentor).Select(x => x.Id);
+
+            var mentors = internship?.Users.Where(x => roles.Contains(x.Id)).ToList();          
+
+            return mentors;
```
Comment (romanstat) [inline InternshipBE/DAL/Repositories/UserRepository.cs]: can be simplified

File: InternshipBE/DAL/Repositories/UserRepository.cs [hunk 10]
```diff
@@ -1,13 +1,31 @@
...
-	}
+    {
+
+        public UserRepository(ApplicationDbContext context) : base(context)
+        {
+        }
+
+        public async Task<List<User>> GetMentorsByInternshipIdAsync(int id)
+        {
+            var internship = await _context.Internships.AsNoTracking().Include(x => x.Users).FirstOrDefaultAsync(x => x.Id == id);
```
Comment (romanstat) [inline InternshipBE/DAL/Repositories/UserRepository.cs]: well, each method wrap to a new line

**Review by romanstat — APPROVED** [review]

File: InternshipBE/WebApi/Controllers/UserController.cs [hunk 11]
```diff
@@ -24,5 +25,11 @@ public async Task<IActionResult> GetUserInfo()
 
             return Ok(await _userService.GetUserInfoByUserNameAsync(userName));
         }
+
+        [HttpGet("getMentorsByInternshipId")]
```
Comment (romanstat) [inline InternshipBE/WebApi/Controllers/UserController.cs]: getSpecificUsersByInternshipId

File: InternshipBE/WebApi/Controllers/UserController.cs [hunk 12]
```diff
@@ -24,5 +25,11 @@ public async Task<IActionResult> GetUserInfo()
...
+        public async Task<IActionResult> GetMentorsByInternshipId([FromQuery] GetSpecificUsersRequest usersRequest)
```
Comment (romanstat) [inline InternshipBE/WebApi/Controllers/UserController.cs]: GetSpecificUsersByInternshipId

**Review by ihhyy — APPROVED** [review]

**Review by Evgeniy-Sinicin — APPROVED** [review]

**Pull request merged**